### PR TITLE
Add flag for accessing auto-downloading binary

### PR DESCRIPTION
**Pull request by abdulla-ashurov**
Add a flag for auto-upgrade

### CODE REVIEW

File: docs/pool-upgrade-how-to.md [hunk 1]
```diff
@@ -160,6 +169,8 @@ application version:
        ```bash
        sudo chown $(whoami) $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
        sudo chmod a+x $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
+       ```
+       > **_Note:_** If you use `auto-upgrade` with `manual-update`, then `manual-update` will work and `auto-upgrade` will be ignored.
```
Comment (uabjabborov) [inline docs/pool-upgrade-how-to.md:173]: Very short and confusing. Please options more explicit:
When `auto-download` is enabled
    a) Binary is put manually to the correct path
       ...
    b) Binary is not there
       ...
Also make sure that propese plan should contain the link and the checksum to the binary according the format specified in cosmovisor docs

File: docs/pool-upgrade-how-to.md [hunk 2]
```diff
@@ -98,68 +98,85 @@ application version:
    steps *(these must be done for all the nodes: validators, observers, seed
    nodes, sentry nodes)*:
 
-    1. Switches current user to the user on behalf of whom `cosmovisor` service
-       is running:
+   1. Make sure that the `auto-download` parameter is present in `cosmovisor.service` (in the folder - `/etc/systemd/system/`) and if the `auto-download` parameter does not exist, then add this `auto-download` parameter to the section `Service` for working **`auto-download`**.
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply docs/pool-upgrade-how-to.md]: Done! ✅

File: docs/pool-upgrade-how-to.md [hunk 3]
```diff
@@ -98,68 +98,85 @@ application version:
...
+      
+      For example:
 
-       ```bash
-       su - <USERNAME>
-       ```
+      ```bash
+      Environment="DAEMON_ALLOW_DOWNLOAD_BINARIES=true"
+      ```
+      > **_Note:_**  In this case, `true` means that the **`auto-download`** is working. If you want to disable **`auto-download`**, you can set the value `false` (`DAEMON_ALLOW_DOWNLOAD_BINARIES=false`) or remove the above line of code from the `cosmovisor.service`.
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply docs/pool-upgrade-how-to.md]: Done! ✅

File: docs/pool-upgrade-how-to.md [hunk 4]
```diff
@@ -98,68 +98,85 @@ application version:
...
+   1. Make sure that the `auto-download` parameter is present in `cosmovisor.service` (in the folder - `/etc/systemd/system/`) and if the `auto-download` parameter does not exist, then add this `auto-download` parameter to the section `Service` for working **`auto-download`**.
+      
+      For example:
 
-       ```bash
-       su - <USERNAME>
-       ```
+      ```bash
+      Environment="DAEMON_ALLOW_DOWNLOAD_BINARIES=true"
+      ```
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply docs/pool-upgrade-how-to.md:107]: Done! ✅

File: docs/pool-upgrade-how-to.md [hunk 5]
```diff
@@ -98,68 +98,85 @@ application version:
...
+
+      ```bash
+      sudo chown $(whoami) $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
+      sudo chmod a+x $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
+      ```
 
-       ```bash
-       sudo chown $(whoami) $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
-       sudo chmod a+x $HOME/.dcl/cosmovisor/upgrades/vX.X.X/bin/dcld
+      When `auto-download` is enabled:
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply docs/pool-upgrade-how-to.md:179]: Done! ✅

File: docs/pool-upgrade-how-to.md [hunk 6]
```diff
@@ -98,68 +98,90 @@ application version:
...
+      ```bash
+      sudo systemctl restart cosmovisor
+      ```
+
+   3. Switches current user to the user on behalf of whom `cosmovisor` service
+      is running:
+
+      ```bash
+      su - <USERNAME>
+      ```
```
Comment (uabjabborov) [inline docs/pool-upgrade-how-to.md:120]: Step 2 is required only if auto-download parameter has changed. So better to integrate it into step 1
It is better to keep "3. Switches current user to the user on behalf of whom `cosmovisor` service" as the 1st step

Comment (abdulla-ashurov) [reply docs/pool-upgrade-how-to.md]: Done! ✅

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**